neutron/local_test.go

```diff
@@ -0,0 +1,92 @@
+package neutron_test
+
+import (
+	"log"
+
+	gc "gopkg.in/check.v1"
+
+	"gopkg.in/goose.v1/client"
+	"gopkg.in/goose.v1/identity"
+	"gopkg.in/goose.v1/neutron"
+	"gopkg.in/goose.v1/testservices"
+	"gopkg.in/goose.v1/testservices/hook"
+	"gopkg.in/goose.v1/testservices/openstackservice"
+)
+
+func registerLocalTests() {
+	gc.Suite(&localLiveSuite{})
+}
+
+// localLiveSuite runs tests from LiveTests using a fake
+// neutron server that runs within the test process itself.
+type localLiveSuite struct {
+	LiveTests
+	openstack             *openstackservice.Openstack
+	retryErrorCount       int  // The current retry error count.
+	retryErrorCountToSend int  // The number of retry errors to send.
+	noMoreIPs             bool // If true, addFloatingIP will return ErrNoMoreFloatingIPs
+	ipLimitExceeded       bool // If true, addFloatingIP will return ErrIPLimitExceeded
+}
+
+func (s *localLiveSuite) SetUpSuite(c *gc.C) {
+	c.Logf("Using identity and neutron service test doubles")
+
+	// Set up an Openstack service.
+	s.cred = &identity.Credentials{
+		User:       "fred",
+		Secrets:    "secret",
+		Region:     "some region",
+		TenantName: "tenant",
+	}
+	var logMsg []string
+	s.openstack, logMsg = openstackservice.New(s.cred, identity.AuthUserPass, false)
+	for _, msg := range logMsg {
+		c.Logf(msg)
+	}
+	s.openstack.UseNeutronNetworking()
+
+	s.openstack.SetupHTTP(nil)
+	s.LiveTests.SetUpSuite(c)
+}
+
+func (s *localLiveSuite) TearDownSuite(c *gc.C) {
+	s.LiveTests.TearDownSuite(c)
+	s.openstack.Stop()
+}
+
+func (s *localLiveSuite) SetUpTest(c *gc.C) {
+	s.retryErrorCount = 0
```

Comment:
You've caught me doing a copy, paste, modify from the nova tests to create the neutron ones and not catching all of the clean up work.  I removed code related to retryErrorCount and retryErrorCountToSend.